src/sqlfluff/core/rules/std/L043.py

```diff
@@ -0,0 +1,264 @@
+"""Implementation of Rule L043."""
+from collections import defaultdict
+from typing import Dict, List, NamedTuple, Optional, Union
+
+from cached_property import cached_property
+
+from sqlfluff.core.dialects.dialect_ansi import AliasInfo
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.rules.base import BaseCrawler, LintResult
+from sqlfluff.core.parser.segments.base import BaseSegment
+from sqlfluff.core.rules.std import L020
+
+
+class RuleFailure(Exception):
+    """Exception class for reporting lint failure inside deeply nested code."""
+
+    pass
+
+
+class WildcardInfo(NamedTuple):
+    """Structure returned by SelectInfo.get_wildcard_info().
+
+    This class is a wrapper for L020.Rule_L020.get_select_statement_info(), plus
+    some additional functionality required by this rule.
+    """
+
+    segment: BaseSegment
+    tables: List[str]
+
+
+class SelectInfo:
```

Comment:
Someday, I could imagine this class and the stuff from L020 moving to a core "query analysis" area of SQLFluff, independent of any particular rules. No rush on this, but an interesting possibility.